### ISSUE TITLE
[NVRHI] stencil shadows don't work

### ISSUE THREAD

**Issue by RobertBeckebans** [RobertBeckebans/RBDOOM-3-BFG#650]
Rendering a stencil shadow volume leads to a fatal error when trying to construct the graphics pipeline.

**Comment by RobertBeckebans** [comment]
I wasted 2 evenings and over 7 hours to get stencil shadows working.
I got them somewhat to work with Vulkan but there was also a lot of zfighting like in RBDOOM-3-BFG 1.4.0
DX12 refuses to render the static shadow volumes of the world scene.
I realized that stencil shadows look really awkward with the additional indirect lighting + the soft new SSAO. 

Shadows should be either shadowmapped or raytraced in 2023 so I removed support for the old stencil volume shadows.
